Zend/zend_ast.c

```diff
@@ -441,3 +444,1011 @@ ZEND_API void zend_ast_apply(zend_ast *ast, zend_ast_apply_func fn) {
+				case ZEND_BW_XOR:              BINARY_OP(" ^ ",   150, 150, 151);
+				case ZEND_IS_IDENTICAL:        BINARY_OP(" === ", 170, 171, 171);
+				case ZEND_IS_NOT_IDENTICAL:    BINARY_OP(" !== ", 170, 171, 171);
+				case ZEND_IS_EQUAL:            BINARY_OP(" == ",  170, 171, 171);
+				case ZEND_IS_NOT_EQUAL:        BINARY_OP(" != ",  170, 171, 171);
+				case ZEND_IS_SMALLER:          BINARY_OP(" < ",   180, 181, 181);
+				case ZEND_IS_SMALLER_OR_EQUAL: BINARY_OP(" <= ",  180, 181, 181);
+				case ZEND_POW:                 BINARY_OP(" ** ",  250, 251, 250);
+				case ZEND_BOOL_XOR:            BINARY_OP(" xor ",  40,  40,  41);
+				default:                       ZEND_ASSERT(0);
+			}
+			break;
+		case ZEND_AST_GREATER:                 BINARY_OP(" > ",   180, 181, 181);
+		case ZEND_AST_GREATER_EQUAL:           BINARY_OP(" >= ",  180, 181, 181);
+		case ZEND_AST_AND:                     BINARY_OP(" && ",  130, 130, 131);
+		case ZEND_AST_OR:                      BINARY_OP(" || ",  120, 120, 121);
+		case ZEND_AST_ARRAY_ELEM:
+			if (ast->child[1]) {
+				zend_ast_export_ex(str, ast->child[1], 80, indent);
+				smart_str_appends(str, " => ");
+			}
+			zend_ast_export_ex(str, ast->child[0], 80, indent);
+			break;
+		case ZEND_AST_NEW:
+			smart_str_appends(str, "new ");
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			smart_str_appendc(str, '(');
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_INSTANCEOF: BINARY_OP(" instanceof ", 230, 231, 231);
+		case ZEND_AST_YIELD:
+			if (priority > 70) smart_str_appendc(str, '(');
+			smart_str_appends(str, "yield ");
+			if (ast->child[0]) {
+				if (ast->child[1]) {
+					zend_ast_export_ex(str, ast->child[1], 70, indent);
+					smart_str_appends(str, " => ");
+				}
+				zend_ast_export_ex(str, ast->child[0], 70, indent);
+			}
+			if (priority > 70) smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_COALESCE: BINARY_OP(" ?? ", 110, 111, 110);
+		case ZEND_AST_STATIC:
+			smart_str_appends(str, "static $");
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			APPEND_DEFAULT_VALUE(1);
+		case ZEND_AST_WHILE:
+			smart_str_appends(str, "while (");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, ") {\n");
+			zend_ast_export_stmt(str, ast->child[1], indent + 1);
+			zend_ast_export_indent(str, indent);
+			smart_str_appendc(str, '}');
+			break;
+		case ZEND_AST_DO_WHILE:
+			smart_str_appends(str, "do {\n");
+			zend_ast_export_stmt(str, ast->child[0], indent + 1);
+			zend_ast_export_indent(str, indent);
+			smart_str_appends(str, "} while (");
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_IF_ELEM:
+			if (ast->child[0]) {
+				smart_str_appends(str, "if (");
+				zend_ast_export_ex(str, ast->child[0], 0, indent);
+				smart_str_appends(str, ") ");
+			} else {
+				smart_str_appends(str, " else ");
+			}
```

Comment:
Thanks for the quick fixes :)

About the `elseif`: I didn't test it, but I suspect that right now it will convert

    if ($a) {
    } elseif ($b) {
    }

into

    if ($a) {
    } if ($b) { // <-- note the missing else
    }

Which would be different.